lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex

```diff
@@ -90,25 +55,23 @@ defmodule LambdaEthereumConsensus.ForkChoice do
         :telemetry.execute([:sync, :on_block], %{slot: slot})
         Logger.info("[Fork choice] Added new block", slot: slot, root: block_root)
 
-        Task.async(__MODULE__, :recompute_head, [new_store])
+        Task.start_link(__MODULE__, :recompute_head, [new_store])
```

Comment:
I'd just remove this level of concurrency for now. Let's just measure how long it takes and show it on grafana, but do it synchronously. 

On a design note: let's never use Task.start_link if we're not using it on a static supervisor (e.g. setting up processes for the first time). For this cases, we want a [dynamic supervisor](https://hexdocs.pm/elixir/1.12/Task.html#module-dynamically-supervised-tasks). In such a case, we would add a `Task.Supervisor` called `RecomputeHeadSupervisor` inside the beacon node supervisor, and we'd call 

```elixir
Task.Supervisor.start_child(
  RecomputeHeadSupervisor, fn -> recompute_head(new_store) end
)
``` 

when we want to recompute the head.

Reply:
Done!